i18n-helpers/src/lib.rs

```diff
@@ -389,14 +421,23 @@ fn heuristic_codeblock<'a>(events: &'a [(usize, Event)]) -> Vec<Group<'a>> {
     };
 
     if is_translate {
-        vec![Group::Translate(events.into())]
+        (
+            vec![Group::Translate {
+                events: events.into(),
+                comment: std::mem::take(&mut ctx.comments).join(" "),
```

Comment:
I think this pattern occurs twice: what about making it a method on the `GroupingContext`?